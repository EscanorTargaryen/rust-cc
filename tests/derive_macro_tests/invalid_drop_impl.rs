use rust_cc::*;

#[derive(Trace)]
struct MyStruct {
    a: (),
}

impl Finalize for MyStruct {
}

impl Drop for MyStruct {
    fn drop(&mut self) {
    }
}

#[derive(Trace)]
enum MyEnum {
    A(),
    B(),
}

impl Finalize for MyEnum {
}

impl Drop for MyEnum {
    fn drop(&mut self) {
    }
}

fn main() {
    fn test<T: Trace>(_t: T) {
    }

    test(MyStruct {
        a: (),
    });
    test(MyEnum::A());
}
