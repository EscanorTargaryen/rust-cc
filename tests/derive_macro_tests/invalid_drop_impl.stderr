error[E0119]: conflicting implementations of trait `Drop` for type `MyStruct`
  --> tests/derive_macro_tests/invalid_drop_impl.rs:3:10
   |
3  | #[derive(Trace)]
   |          ^^^^^ conflicting implementation for `MyStruct`
...
11 | impl Drop for MyStruct {
   | ---------------------- first implementation here
   |
   = note: this error originates in the derive macro `Trace` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `Drop` for type `MyEnum`
  --> tests/derive_macro_tests/invalid_drop_impl.rs:16:10
   |
16 | #[derive(Trace)]
   |          ^^^^^ conflicting implementation for `MyEnum`
...
25 | impl Drop for MyEnum {
   | -------------------- first implementation here
   |
   = note: this error originates in the derive macro `Trace` (in Nightly builds, run with -Z macro-backtrace for more info)
